Scripts/RemediationScripts/Remediate-SetClusterProtectionLevelForServiceFabric.ps1

```diff
@@ -216,28 +235,94 @@ function Set-ClusterProtectionLeveltoEncryptandSignforServiceFabric
+            	
+            Write-Host "No Service Fabric(s) found in input json file for remediation." -ForegroundColor $([Constants]::MessageType.Error)	
+            Write-Host $([Constants]::SingleDashLine)	
+            return	
+        
+        }
+         $validResources | ForEach-Object { 	
+            try	
+            {
+            $ServiceFabricResource =  Get-AzServiceFabricCluster -ResourceGroupName $_.ResourceGroupName -Name $_.ResourceName -ErrorAction SilentlyContinue
+            $ServiceFabricDetails += $ServiceFabricResource  | Select-Object @{N='ResourceId';E={$_.Id}},
+                                                                          @{N='ResourceGroupName';E={$_.Id.Split("/")[4]}},
+                                                                          @{N='ResourceName';E={$_.Name}}, 
+                                                                          @{N='IsClusterProtectionLevelExist';E={$_.FabricSettings.Parameters.Name -Contains 'ClusterProtectionLevel'}},
+                                                                          @{N='ClusterProtectionLevelValue';E={if(($_.FabricSettings.Parameters.Name -Contains 'ClusterProtectionLevel') -eq $true)
+                                                                          { 
+                                                                                $param = $_.FabricSettings | Where-Object {$_.Parameters.Name -eq 'ClusterProtectionLevel'}
+                                                                                $param.Parameters.Value
+                                                                          }                                                                          
+                                                                          }},
+                                                                          @{N='NodeVMCount';E={$_.NodeTypes.VmInstanceCount}}
+             }
+            catch
+            {
+                Write-Host "Valid resource id(s) not found in input json file. Error: [$($_)]" -ForegroundColor $([Constants]::MessageType.Error)	
+                Write-Host "Skipping the Resource:  [$($_.ResourceName)]..."
+                $logResource = @{}
+                 $logResource.Add("ResourceGroupName",($_.ResourceGroupName))	
+                $logResource.Add("ResourceName",($_.ResourceName))	
+                $logResource.Add("Reason","Valid resource id(s) not found in input json file.")    	
+                $logSkippedResources += $logResource	
+                Write-Host $([Constants]::SingleDashLine)
+                Write-Host "Error fetching Service Fabric(s) resource: Resource ID:  $($resourceId). Error: $($_)" -ForegroundColor $([Constants]::MessageType.Error)
+             }	
+        }	
+    }
+    else{	
+        # No file path provided as input to the script. Fetch all Service Fabric(s) in the Subscription.                                                                  
```

Comment:
There is no code in this else block.

The below code related to file input should be inside it. Check once.